## HW1 Sentiment Analysis

<b>List1-Tweet</b>

In [1]:
%autosave 120

Autosaving every 120 seconds


In [2]:
tweets = []
f = open('Trump.txt','r')

for w in f:
    w = w.lower().strip()
    w = w.replace('@','').replace(':','').replace('.','').replace('*','').\
        replace(',','').replace('rt','').replace('?','').replace('&','').replace('\\','')
    for w in w.split():
        tweets.append(w)
tweets[:20]
len(tweets)
f.close()

In [3]:
print(tweets[:20])

["b'", 'realdonaldtrump', 'i', 'have', 'fully', 'authorized', 'the', 'total', 'declassification', 'of', 'any', 'amp;', 'all', 'documents', 'peaining', 'to', 'the', 'single', 'greatest', "politicalxe2x80xa6n'b'"]


 <b>List2-Positive</b>

In [4]:
positive = []
f = open('positive.txt','r')
for w in f: 
    w = w.lower().strip()
    positive.append(w)
len(positive)
positive[:20]
f.close()

In [5]:
print(positive[:20])

['a+', 'abound', 'abounds', 'abundance', 'abundant', 'accessable', 'accessible', 'acclaim', 'acclaimed', 'acclamation', 'accolade', 'accolades', 'accommodative', 'accomodative', 'accomplish', 'accomplished', 'accomplishment', 'accomplishments', 'accurate', 'accurately']


<b>List 3 - Negative</b>

In [6]:
negative = []
f = open('negative.txt','r')
for w in f:
    w = w.lower()
    w = w.strip()
    negative.append(w)
len(negative)
negative[:20]
f.close()

In [7]:
print(negative[:20])

['2-faced', '2-faces', 'abnormal', 'abolish', 'abominable', 'abominably', 'abominate', 'abomination', 'abort', 'aborted', 'aborts', 'abrade', 'abrasive', 'abrupt', 'abruptly', 'abscond', 'absence', 'absent-minded', 'absentee', 'absurd']


<b>List 4 - Stopword</b>

In [8]:
stopword = []
f = open('stopwords.txt','r')
for w in f:
    w = w.lower().strip()
    stopword.append(w)
len(stopword)
stopword[:20]
f.close()

In [9]:
print(stopword[:20])

['a', 'about', 'above', 'across', 'after', 'again', 'against', 'all', 'almost', 'alone', 'along', 'already', 'also', 'although', 'always', 'among', 'an', 'and', 'another', 'any']


<b>Compare the words in tweets with (positive, negative and stopword)</b>

In [10]:
count_p = 0
count_n = 0
count_s = 0
count_other = 0
count_total = 0

In [11]:
for i in tweets:
    count_total += 1
    if i in positive:
        count_p += 1
    elif i in negative:
        count_n += 1
    elif i in stopword:
        count_s += 1
    else : count_other +=1

In [14]:
print ("positive words are %d" %count_p)
print ("negative words are %d" %count_n)
print ("stop words are %d" %count_s)
print ("other words are %d" %count_other)
print ("total word count is %d" %count_total )
print ('*' * 90)
print ("postive ratio is %.2f" %(count_p/count_total))
print ("negative ratio is %.2f" %(count_n/count_total))
print ("stop ratio is %.2f" %(count_s/count_total))
print ("other ratio is %.2f" %(count_other/count_total))
print ("positive vs negative ratio is %.2f" % (count_p/count_n))

positive words are 74392
negative words are 50562
stop words are 625227
other words are 729914
total word count is 1480095
******************************************************************************************
postive ratio is 0.05
negative ratio is 0.03
stop ratio is 0.42
other ratio is 0.49
positive vs negative ratio is 1.47


## Conclusion 
The trump word count is strongly positive. The positive ratio, 0.05 is larger than the negative ratio, 0.03, and the positive over negative ratio is 1.47 > 1. 

## #2**Tweets keyword of Brady**

<b>List1-Tweet</b>

In [15]:
btweets = []
f = open('Brady.txt','r')

for w1 in f:
    w1 = w1.lower().strip()
    w1 = w1.replace('@','').replace(':','').replace('.','').replace('#','').replace('*','').\
        replace(',','').replace('http:s//','').replace('rt','').replace('?','').\
        replace('&','').replace('/','').replace('https//','').replace('\\','').\
        replace('//','').replace('http','')
    for w1 in w1.split():
        btweets.append(w1)
btweets[:20]
len(tweets)
f.close()

In [16]:
print(btweets[:20])

["b'", 'cornerpubron', 'three', 'constants', 'in', 'lifenn1', 'deathn2', 'taxesn3', 'nick', 'folesxe2x80x99', 'space', 'in', 'tom', 'bradyxe2x80x99s', 'headnntnf', 'tbvschi', 'bears', "stcof5xe2x80xa6n'b'", 'maggiehendricks', 'ixe2x80x99m']


 <b>List2-Positive</b>

In [17]:
positive = []
f = open('positive.txt','r')
for w1 in f: 
    w1 = w1.lower().strip()
    positive.append(w1)
len(positive)
positive[:20]
f.close()

In [18]:
print(positive[:20])

['a+', 'abound', 'abounds', 'abundance', 'abundant', 'accessable', 'accessible', 'acclaim', 'acclaimed', 'acclamation', 'accolade', 'accolades', 'accommodative', 'accomodative', 'accomplish', 'accomplished', 'accomplishment', 'accomplishments', 'accurate', 'accurately']


<b>List 3 - Negative</b>

In [26]:
negative = []
f = open('negative.txt','r')
for w1 in f:
    w1 = w1.lower()
    w1 = w1.strip()
    negative.append(w1)
len(negative)
negative[:20]
f.close()

In [27]:
print(negative[:20])

['2-faced', '2-faces', 'abnormal', 'abolish', 'abominable', 'abominably', 'abominate', 'abomination', 'abort', 'aborted', 'aborts', 'abrade', 'abrasive', 'abrupt', 'abruptly', 'abscond', 'absence', 'absent-minded', 'absentee', 'absurd']


<b>List 4 - Stopword</b>

In [28]:
stopword = []
f = open('stopwords.txt','r')
for w1 in f:
    w1 = w1.lower()
    w1 = w1.strip()
    stopword.append(w1)
len(stopword)
stopword[:20]
f.close()

In [29]:
print(stopword[:20])

['a', 'about', 'above', 'across', 'after', 'again', 'against', 'all', 'almost', 'alone', 'along', 'already', 'also', 'although', 'always', 'among', 'an', 'and', 'another', 'any']


In [33]:
count_p1 = 0
count_n1 = 0
count_s1 = 0
count_other1 = 0
count_total1 = 0

In [34]:
for i in btweets:
    count_total1 += 1
    if i in positive:
        count_p1 += 1
    elif i in negative:
        count_n1 += 1
    elif i in stopword:
        count_s1 += 1
    else : count_other1 +=1

In [35]:
print ("positive words = %d" %count_p1)
print ("negative words = %d" %count_n1)
print ("stop words = %d" %count_s1)
print ("other words = %d" %count_other1)
print ("total word count = %d" %count_total1)
print ('*' * 90)
print ("postive ratio = %.2f" %(count_p1/count_total1))
print ("negative ratio = %.2f" %(count_n1/count_total1))
print ("stop ratio = %.2f" %(count_s1/count_total1))
print ("other ratio = %.2f" %(count_other1/count_total1))
print ("positive vs negative ratio is %.2f" % (count_p1/count_n1))

positive words = 154
negative words = 192
stop words = 2857
other words = 3924
total word count = 7127
******************************************************************************************
postive ratio = 0.02
negative ratio = 0.03
stop ratio = 0.40
other ratio = 0.55
positive vs negative ratio is 0.80


## Conclusion

The Brady word count is weakly negative. The positive ratio, 0.02 which is sightly smaller than negative ratio, 0.03. Then, the positive vs negative ratio 0.8 is slightly < 1.